lldb/packages/Python/lldbsuite/test/cpu_feature.py

```diff
@@ -0,0 +1,58 @@
+"""
+Platform-agnostic helper to query for CPU features.
+"""
+
+import re
+
+
+class CPUFeature:
+    def __init__(self, linux_cpu_info_flag: str, darwin_sysctl_key: str):
+        self.cpu_info_flag = linux_cpu_info_flag
+        self.sysctl_key = darwin_sysctl_key
+
+    def __str__(self):
+        return self.cpu_info_flag
+
+    def is_supported(self, triple, cmd_runner):
+        if re.match(".*-.*-linux", triple):
+            err_msg, res = self._is_supported_linux(cmd_runner)
+        elif re.match(".*-apple-.*", triple):
```

Comment:
Are there also -darwin- triples? I have no idea but you could look for existing checks in other tests.

Reply:
Also note: My aim is to preserve the existing behavior "NFC".  We can refine/adapt in follow-ups as necessary.